### PR TITLE
Use flutter_simple_treeview

### PR DESCRIPTION
**Pull request by domesticmouse**
Simplifying the native sample by replacing included code with a published widget. Also adds scroll bars.

### CODE REVIEW

File: experimental/desktop_photo_search/lib/src/widgets/photo_details.dart [hunk 1]
```diff
@@ -29,96 +29,101 @@ class PhotoDetails extends StatefulWidget {
...
-    return RichText(
-      text: TextSpan(
-        style: Theme.of(context).textTheme.bodyText2,
-        children: [
-          const TextSpan(text: 'Photo by '),
-          TextSpan(
-            text: widget.photo.user.name,
-            style: const TextStyle(
-              decoration: TextDecoration.underline,
+    return Expanded(
```
Comment (RedBrogdon) [inline experimental/desktop_photo_search/lib/src/widgets/photo_details.dart:34]: I've run into trouble including `Expanded` at the top of build methods like this, and I'd recommend moving it out of this widget so it's built as part of the parent's code.

The reason is that Expanded (which is a Flexible) is meant to work with a Flex ancestor, and there are rules about what can come in between:

https://github.com/flutter/flutter/blob/74a2e6a5e06332ae8689b077b5c37f4213fde997/packages/flutter/lib/src/widgets/basic.dart#L4374

By adding just the Expanded here, there's now a subtle requirement to using the PhotoDetail widget ("You need to put this pretty directly into a Flex or it won't work"). In this app's case, the `Split` widget higher in the tree is building a Flex and putting this in it, so the code works. But in another situation, this widget could start throwing exceptions.

File: experimental/desktop_photo_search/lib/main.dart [hunk 2]
```diff
@@ -11,10 +11,10 @@ import 'package:logging/logging.dart';
 import 'package:menubar/menubar.dart' as menubar;
 import 'package:meta/meta.dart';
 import 'package:provider/provider.dart';
+import 'package:flutter_simple_treeview/flutter_simple_treeview.dart';
```
[the remark this answers is not in the document]
Comment (domesticmouse) [reply experimental/desktop_photo_search/lib/main.dart:14]: I thought there was a lint for this sort of thing, but I can't find it. Ideas?

File: experimental/desktop_photo_search/lib/src/model/photo_search_model.dart [hunk 3]
```diff
@@ -4,45 +4,48 @@
...
-  }
-}
-
-class _SearchEntry extends Entry {
-  _SearchEntry(String query, List<Photo> photos, PhotoSearchModel model)
-      : super(
-          query,
-          List<Entry>.unmodifiable(
-            photos.map<Entry>((photo) => _PhotoEntry(photo, model)),
+TreeNode _searchEntry(
```
[the remark this answers is not in the document]
Comment (domesticmouse) [reply experimental/desktop_photo_search/lib/src/model/photo_search_model.dart]: Moved `TreeNode` construction to `main.dart`, the model is now pure again.

File: experimental/desktop_photo_search/lib/src/widgets/photo_details.dart (hunk 1 above)
Comment (domesticmouse) [reply experimental/desktop_photo_search/lib/src/widgets/photo_details.dart:34]: This is an internal helper method for the main `build` method, which is below. Should I re-arrange the order of methods to make this more obvious?

File: experimental/desktop_photo_search/lib/main.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (RedBrogdon) [reply experimental/desktop_photo_search/lib/main.dart]: This one, I think: https://dart-lang.github.io/linter/lints/directives_ordering.html

Is that in our standard analysis options? If not, I'd vote for it.

File: experimental/desktop_photo_search/lib/src/widgets/photo_details.dart (hunk 1 above)
Comment (RedBrogdon) [reply experimental/desktop_photo_search/lib/src/widgets/photo_details.dart:34]: 🤦 

I think I had so much trouble with Expanded early on in my Flutter journey that I still bear scars. I saw `return Expanded` and some part of my brain thought "I must warn the village!".

You're right that code's fine as-is. If it were me, I think I'd move it just out of personal preference, but it's definitely just a preference.

**Comment by domesticmouse** [comment]
PTAL @johnpryan @RedBrogdon

**Pull request merged**